### DISCUSSION TITLE
Capture to file in local directory

### DISCUSSION THREAD

**Discussion by cshyshtof** [chhoumann/quickadd#214] [Q&A]
I have many projects (directories) which contain separate files for project info, meetings and tasks. I would like to create new tasks in Project Tasks file from some meeting notes. Is it possible to capture highlighted text and add it as task to a file which is in current directory? When I add a capture to that file name it creates the new one in root directory. Each project has different name and each project has the file for tasks named the same Project Tasks. How can I do that, as in QuickAdd I can define static directory only or use formats which cover mainy dates. I do not want to ask for file name, I want to automatically add to file located in current (not root) directory. Is it possible?

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that.

Yes, this works - there's no built-in "active file's folder" token, but a two-step macro gets you exactly this with no prompting:

1. Create a Macro choice. Its first command is a small user script that grabs the active file's folder:

```js
module.exports = async (params) => {
	const file = params.app.workspace.getActiveFile();
	if (!file) throw new Error("No active file");
	params.variables.projectFolder = file.parent?.path ?? "";
};
```

2. The second command is a Capture choice with *Capture To* set to:

```
{{VALUE:projectFolder}}/Project Tasks.md
```

Variables set by a macro script resolve in later steps without prompting, so the capture goes straight to the `Project Tasks` file sitting next to whatever note you're in.

3. For the highlighted text, use `{{SELECTED}}` in the capture format, e.g.:

```
- [ ] {{SELECTED}}
```

Run the macro from your meeting note and the selection lands as a task in that project's own `Project Tasks.md`, no questions asked.

Docs: https://quickadd.obsidian.guide/docs/Choices/MacroChoice

Closing as resolved.
